core/src/avm2/value.rs

```diff
@@ -0,0 +1,262 @@
+//! AVM2 values
+
+use crate::avm1::AvmString;
+use crate::avm2::names::Namespace;
+use crate::avm2::object::Object;
+use crate::avm2::script::TranslationUnit;
+use crate::avm2::Error;
+use gc_arena::{Collect, MutationContext};
+use std::f64::NAN;
+use swf::avm2::types::{DefaultValue as AbcDefaultValue, Index};
+
+/// An AVM2 value.
+///
+/// TODO: AVM2 also needs Scope, Namespace, and XML values.
+#[derive(Clone, Collect, Debug)]
+#[collect(no_drop)]
+pub enum Value<'gc> {
+    Undefined,
+    Null,
+    Bool(bool),
+    Number(f64),
+    String(AvmString<'gc>),
+    Namespace(Namespace<'gc>),
+    Object(Object<'gc>),
+}
```

Comment:
I have no strong opinion either way.

I did use `Value` specifically because it matches the AVM1 `Value`. I doubt we're going to have people used to hacking on avmplus wondering where `Atom` is here, as much as we're going to have people used to hacking on Ruffle's AVM1 interpreter wondering where `Value` is.